src/core/crypto/ecdsa.cpp

```diff
@@ -53,6 +59,36 @@ namespace Ecdsa {
 
 Error P256::KeyPair::Generate(void)
 {
+#if defined(MBEDTLS_USE_TINYCRYPT)
+
+    mbedtls_pk_context    pk;
+    mbedtls_uecc_keypair *keypair;
+    int                   ret;
+
+    mbedtls_pk_init(&pk);
+
+    ret = mbedtls_pk_setup(&pk, mbedtls_pk_info_from_type(MBEDTLS_PK_ECKEY));
+    VerifyOrExit(ret == 0);
+
+    keypair = mbedtls_pk_uecc(pk);
+
+    ret = uECC_make_key(keypair->public_key, keypair->private_key);
+    VerifyOrExit(ret == UECC_SUCCESS);
+
+    ret = mbedtls_pk_write_key_der(&pk, mDerBytes, sizeof(mDerBytes));
+    VerifyOrExit(ret > 0);
+
+    mDerLength = static_cast<uint8_t>(ret);
+
+    memmove(mDerBytes, mDerBytes + sizeof(mDerBytes) - mDerLength, mDerLength);
+
+exit:
+    mbedtls_pk_free(&pk);
+
+    return (ret >= 0) ? kErrorNone : MbedTls::MapError(ret);
```

Comment:
This method and the ones belwo are repeating a good chunk of the same code under `MBEDTLS_USE_TINYCRYPT`. Suggest reusing the code and only changing the parts that are different under `TINYCRYPT`.  Here is an example for `Generate()`:

```c++
Error P256::KeyPair::Generate(void)
{
    mbedtls_pk_context pk;
    int                ret;

    mbedtls_pk_init(&pk);

    ret = mbedtls_pk_setup(&pk, mbedtls_pk_info_from_type(MBEDTLS_PK_ECKEY));
    VerifyOrExit(ret == 0);

#ifndef MBEDTLS_USE_TINYCRYPT
    ret = mbedtls_ecp_gen_key(MBEDTLS_ECP_DP_SECP256R1, mbedtls_pk_ec(pk), MbedTls::CryptoSecurePrng, nullptr);
    VerifyOrExit(ret == 0);
#else
    {
        mbedtls_uecc_keypair *keypair= mbedtls_pk_uecc(pk);

        ret = uECC_make_key(keypair->public_key, keypair->private_key);
        VerifyOrExit(ret == UECC_SUCCESS);
    }
#endif

    ret = mbedtls_pk_write_key_der(&pk, mDerBytes, sizeof(mDerBytes));
    VerifyOrExit(ret > 0);

    mDerLength = static_cast<uint8_t>(ret);

    memmove(mDerBytes, mDerBytes + sizeof(mDerBytes) - mDerLength, mDerLength);

exit:
    mbedtls_pk_free(&pk);

    return (ret >= 0) ? kErrorNone : MbedTls::MapError(ret);
}
```